sunkit_spex/fitting/examples/simple_fitting.py

```diff
@@ -0,0 +1,304 @@
+SAVE_FIG = False
+# this should keep "random" stuff is the same ech run
+np.random.seed(seed=10)
+
+# get some fake photon energies
+
+
+def photon_energies(start, stop, inc):
+    """ Get a `ndarray` of energies. """
+    return np.arange(start, stop, inc)
+
+
+def plot_fake_photon_spectrum(axis,
+                              photon_energies,
+                              photon_values,
+                              title="Fake Photon Spectrum"):
+    """ Plots the fake photon spectrum. """
+
+    axis.plot(photon_energies, photon_values)
+    axis.set_xlabel("Energy [keV]")
+    axis.set_ylabel("ph s$^{-1}$ cm$^{-2}$ keV$^{-1}$")
+    axis.set_title(title)
+
+
+def response_matrix(photon_energies):
+    """
+    Generate a square matrix with off-diagonal terms to mimic an
+    instrument response matrix.
+    """
+    # fake SRM
+    fake_srm = np.identity(photon_energies.size)
+
+    # add some off-diagonal terms
+    for c, r in enumerate(fake_srm):
+        # add some features into the fake SRM
+        off_diag = np.random.rand(c)*0.005
+
+        # add a diagonal feature
+        _x = 50
+        if c >= _x:
+            off_diag[-_x] = np.random.rand(1)[0]
+
+        # add a vertical feature in
+        _y = 200
+        __y = 30
+        if c > _y+100:
+            off_diag[_y-__y//2:_y+__y//2] = np.arange(2*(__y//2))*np.random.rand(2*(__y//2))*5e-4
+
+        # put these features in the fake_srm row and normalize
+        r[:off_diag.size] = off_diag
+        r /= np.sum(r)
+
+    return fake_srm
+
+
+def plot_fake_srm(axis, fake_srm, photon_energies, title="Fake SRM"):
+    """ Plots the fake response matrix. """
+
+    axis.imshow(fake_srm,
+                origin="lower",
+                extent=[photon_energies[0],
+                        photon_energies[-1],
+                        photon_energies[0],
+                        photon_energies[-1]],
+                norm=LogNorm())
+    axis.set_ylabel("Photon Energies [keV]")
+    axis.set_xlabel("Count Energies [keV]")
+    axis.set_title(title)
+
+
+def plot_fake_count_spectrum(axis,
+                             photon_energies,
+                             photon_model_features,
+                             gaussian_feature,
+                             total_count_spectrum,
+                             total_count_spectrum_wnoise,
+                             title="Fake Count Spectrum"):
+    """ Plots the fake count spectrum and its componenets. """
+
+    axis.plot(photon_energies,
+              photon_model_features,
+              label="photon model features")
+    axis.plot(photon_energies,
+              gaussian_feature,
+              label="gaussian feature")
+    axis.plot(photon_energies,
+              total_count_spectrum,
+              label="total fake spectrum")
+    axis.plot(photon_energies,
+              total_count_spectrum_wnoise,
+              label="total fake spectrum + noise",
+              lw=0.5)
```

Comment:
I think you should be able to iterate over the model components

Reply:
I thought so too when I wrote this but was too lazy :/

I've re-ordered the example for the docs now (which I think looks and flows way better now) so I don't think this is as much of an issue with the written code now. I could be wrong though!